### PR TITLE
tests: kernel: App to validate boot time page table

### PR DESCRIPTION
**Pull request by akhileshupadhyay**
Testcase developed to validate boot time page table faults.

Signed-off-by: Akhilesh Kumar Upadhyay <akhilesh.kumarx.upadhyay@intel.com>

### CODE REVIEW

**Review by galak — DISMISSED** [review]
This seems like a pretty x86 specific test, can we rename so its clear that its x86 specific.

File: tests/kernel/boot_page_table/src/main.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include <ztest.h>
```
Comment (nashif) [inline tests/kernel/boot_page_table/src/main.c]: missing copyright/license header

File: tests/kernel/boot_page_table/src/boot_page_table.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#ifndef X86_MMU_GET_PT_ADDR
```
Comment (nashif) [inline tests/kernel/boot_page_table/src/boot_page_table.h]: missing copyright/license header

File: tests/kernel/boot_page_table/src/boot_page_table.c [hunk 3]
```diff
@@ -0,0 +1,110 @@
+#include <zephyr.h>
```
Comment (nashif) [inline tests/kernel/boot_page_table/src/boot_page_table.c]: missing copyright/license header

**Review by nashif — CHANGES_REQUESTED** [review]
Can you please add a README explaining what is being tested here and give more details?

File: tests/kernel/boot_page_table/src/boot_page_table.h (hunk 2 above)
Comment (nashif) [reply tests/kernel/boot_page_table/src/boot_page_table.h]: you also need a header guard here

**Comment by akhileshupadhyay** [comment]
Missing copyright/license header is added to the necessary files. README file is added with details. The header guards are also added.

File: tests/kernel/boot_page_table/src/boot_page_table.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#define MMU_READ                  0x00
+#define MMU_WRITE                 0x01
+#define MMU_READ_WRITE             (MMU_READ | MMU_WRITE)
+#define MMU_PAGE_USER             0x02
+#define START_ADDR_RANGE          0x12340000
+#define ADDR_SIZE                 0x1000
+#define STARTING_ADDR_RANGE_LMT   0x0009ff
+#define START_ADR_RANGE_OVRLP_LMT 0x001000
+#define REGION_PERM                (MMU_READ_WRITE | MMU_PAGE_USER)
+#endif
```
Comment (AdithyaBaglody) [inline tests/kernel/boot_page_table/src/boot_page_table.h]: move line 37 (#endif )to line 27

File: tests/kernel/boot_page_table/src/boot_page_table.c [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <mmustructs.h>
+#include <ztest.h>
+#include "boot_page_table.h"
+
+MMU_BOOT_REGION(START_ADDR_RANGE, ADDR_SIZE, REGION_PERM);
```
Comment (AdithyaBaglody) [inline tests/kernel/boot_page_table/src/boot_page_table.c]: add multiple regions into the test. as of now it is testing a single memory region.
Create multiple memory regions and validate them with the below APIs.

**Review by AdithyaBaglody — CHANGES_REQUESTED** [review]

File: tests/kernel/boot_page_table/src/boot_page_table.h (hunk 4 above)
Comment (akhileshupadhyay) [reply tests/kernel/boot_page_table/src/boot_page_table.h]: #endif  moved to line 27

File: tests/kernel/boot_page_table/src/boot_page_table.c (hunk 5 above)
Comment (akhileshupadhyay) [reply tests/kernel/boot_page_table/src/boot_page_table.c]: multiple memory regions has been created and added to validate upon.

File: tests/kernel/boot_page_table/src/boot_page_table.c [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+{
+	zassert_true(before_start_addr_range() == TC_PASS, NULL);
+}
+
+void test_ending_start_addr_range(void)
+{
+	zassert_true(ending_start_addr_range() == TC_PASS, NULL);
+}
+
+void test_starting_addr_range1(void)
```
Comment (AdithyaBaglody) [inline tests/kernel/boot_page_table/src/boot_page_table.c]: Pass arguments into the same function. if arguments== 1 then take start addr 1 if argument == 2 then take start addr2 so on.
Don't duplicate the code.

**Review by AdithyaBaglody — CHANGES_REQUESTED** [review]

File: tests/kernel/boot_page_table/src/boot_page_table.c (hunk 6 above)
Comment (akhileshupadhyay) [reply tests/kernel/boot_page_table/src/boot_page_table.c]: @AdithyaBaglody . The memory address are already passed based on the arguments.

**Review by AdithyaBaglody — APPROVED** [review]

**Comment by akhileshupadhyay** [comment]
@galak Could you please review the changes requested by you.

**Review by kumarvikash1 — APPROVED** [review]
copyright/license header, header guard and line breaking has been addressed. Approved.

**Comment by kumarvikash1** [comment]
@SebastianBoe @galak  : Can you please review the changes. This PR is planned for R-1.10.

**Comment by SebastianBoe** [comment]
I am not able to review, removed myself from reviewers list.

**Comment by galak** [comment]
My comment about renaming since this is x86 specific is still valid.

**Comment by akhileshupadhyay** [comment]
@galak . Could you please check, the renaming is done for x86.

File: tests/kernel/x86boot_page_table/testcase.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+tests:
+-   test:
+        platform_whitelist: qemu_x86
```
Comment (galak) [inline tests/kernel/x86boot_page_table/testcase.yaml]: Is there any reason this test will not run on any x86 platform?

Seems like we should have:

arch_whitelist: x86

for sure in here.

**Comment by akhileshupadhyay** [comment]
@galak . The changes have been addressed for arch_whitelist: x86

**Comment by galak** [comment]
@akhileshupadhyay thanks.  I removed my 'requests changes'.  I leave this up to @nashif  to decide if he's ok with it and wants it merged in 1.10.

**Review by nashif — CHANGES_REQUESTED** [review]
remove x86 from the path name x86boot_page_table/...

**Comment by akhileshupadhyay** [comment]
@nashif Could please check, removed x86 from x86boot_page_table path name.

**Review by nashif — APPROVED** [review]

**Pull request merged**